python_modules/dagster/dagster/_cli/asset.py

```diff
@@ -11,6 +30,57 @@ def asset_cli():
     """
 
 
+@asset_cli.command(name="materialize", help="Execute a run to materialize a selection of assets")
+@python_origin_target_argument
+@click.option("--select", help="Asset selection to target", required=True)
+@click.option("--partition", help="Asset partition to target", required=False)
+def asset_materialize_command(**kwargs):
+    with capture_interrupts():
+        with get_instance_for_service("``dagster asset materialize``") as instance:
+            execute_materialize_command(instance, kwargs)
+
+
+@telemetry_wrapper
+def execute_materialize_command(instance: DagsterInstance, kwargs: Mapping[str, str]) -> None:
+    repository_origin = get_repository_python_origin_from_kwargs(kwargs)
+
+    recon_repo = recon_repository_from_origin(repository_origin)
+    repo_def = recon_repo.get_definition()
+
+    asset_keys = parse_asset_selection(
+        assets_defs=list(repo_def._assets_defs_by_key.values()),
+        source_assets=list(repo_def.source_assets_by_key.values()),
+        asset_selection=kwargs["select"].split(","),
+    )
+
+    implicit_job_def = repo_def.get_implicit_job_def_for_assets(asset_keys)
+    if implicit_job_def is None:
+        raise DagsterInvalidSubsetError(
+            "All selected assets must share the same PartitionsDefinition or have no"
+            " PartitionsDefinition"
```

Comment:
pardon my ignorance, but does this error message map the condition being tested here?

Reply:
Interestingly, yes.

All the assets get smushed into one implicit job - unless they have different PartitionsDefinitions, in which case they can't be.  So if we can't find an implicit job with all the given assets, it's because they couldn't be put into the same implicit job, because of their conflicting PartitionsDefinitions.

Reply:
I'll add a comment